### PR TITLE
[MLIR][LLVM] Add llvm.intr.coro.promise

### PR DESCRIPTION
**Pull request by yonillasky**
Added to allow generating these intrinsics in out-of-tree MLIR passes.

### CODE REVIEW

File: mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir [hunk 1]
```diff
@@ -712,6 +712,13 @@ llvm.func @coro_resume(%arg0: !llvm.ptr) {
   llvm.return
 }
 
+// CHECK-LABEL: @coro_promise
```
Comment (gysit) [inline mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir:715]: Can you add a check that matches the intrinsics declaration at the end of the file, e.g., right after:
`// CHECK-DAG: declare void @llvm.coro.resume(ptr)`

File: mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td [hunk 2]
```diff
@@ -435,6 +435,13 @@ def LLVM_CoroResumeOp : LLVM_IntrOp<"coro.resume", [], [], [], 0> {
   let assemblyFormat = "$handle attr-dict `:` qualified(type($handle))";
 }
 
+def LLVM_CoroPromiseOp : LLVM_IntrOp<"coro.promise", [], [], [], 1> {
+  let arguments = (ins LLVM_AnyPointer:$handle,
+                       I32:$align,
+                       I1:$from);
+  let assemblyFormat = "$handle `,` $align `,` $from attr-dict `:` functional-type(operands, results)";
```
Comment (gysit) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td:442]:
```suggestion
                       I1:$from);
  let results = (outs LLVM_AnyPointer:$res);
  let assemblyFormat = "$handle `,` $align `,` $from attr-dict `:` functional-type(operands, results)";
```
If I understand correctly the result has to be a pointer? If yes, I would specify the result type to ensure it is verified.

**Review by gysit — COMMENTED** [review]
Nice!

Can you add an import test as well? It can be added to the following file:
`llvm-project/mlir/test/Target/LLVMIR/Import/intrinsic.ll`

**Comment by gysit** [comment]
PS the commit message should usually be prefixed with [MLIR] and optionally also with the dialect [MLIR][LLVM].

File: mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir (hunk 1 above)
Comment (yonillasky) [reply mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir:715]: For some reason, none of the llvm.coro intrinsics are listed there.

**Comment by yonillasky** [comment]
> Nice!
> 
> Can you add an import test as well? It can be added to the following file: `llvm-project/mlir/test/Target/LLVMIR/Import/intrinsic.ll`

Just saw this, will push again soon

**Review by gysit — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**